### PR TITLE
test(graphql): add negative test cases for invalid inputs and arguments

### PR DESCRIPTION
**Pull request by sundersc**
#### Description of changes

Add negative test cases for mutations and queries with invalid inputs and arguments.

- Invalid data type check
- Invalid arguments length
- Invalid filter with 'and' and 'or' conditions

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            { title: 'test', description: 'task 1 description', priority: 2, dueDate: 1000000 }, // Invalid date argument
+            { title: 'task 1', description: 'task 1 description', priority: 5.1, dueDate: '2022-01-01T00:00:00.000Z' }, // Passing float in place of int
+            { title: 'task 1', description: 'task 1 description', priority: "invalid", dueDate: '2022-01-01T00:00:00.000Z' }, // Passing string in place of int
+        ];
+        
+        await Promise.all(
+            invalid_create_inputs.map(async (input) => {
+                try {
+                    const createResult = await createTask(input);
+                    expect(false).toBeTruthy();
```
Comment (alharris-at) [inline packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts:46]: Does this become more readable as an `expect(async () => await createTask(input).toThrow(/matcher/)

I'm fine as-is, so treat as a question.

File: packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            { priority: { gt: 1.5 } }, // Invalid datatype - passing float for int
+            { priority: { lt: "test" } }, // Invalid datatype - passing string for int
+            { and: { priority: { lt: "test" } } }, // Invalid datatype - passing string for int with 'and' condition
+            { priority: { lt: "test" } }, // Invalid datatype - passing string for int
+            { priority: { between: [1] } }, // Incorrect number of arguments for between operator
+            { priority: { between: [1, 2.5] } }, // Incorrect argument value for between operator
+            { or: [ { priority: { between: [1, 2.5] } }, { title: { gt: "test" } } ] }, // Incorrect argument value for between operator with 'or' condition
+        ];
+
+        await Promise.all(
```
Comment (alharris-at) [inline packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts:68]: Can you pull this block out to a utlity? Then these tests will just be 'data setup'.test basically.

**Review by alharris-at — DISMISSED** [review]
Non-blocking comments.

File: packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts (hunk 2 above)
Comment (sundersc) [reply packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts]: Makes sense. Moved the logic to a separate method.

File: packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts (hunk 1 above)
Comment (sundersc) [reply packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts]: I've considered this but we will lose the ability to verify error.errors is defined and error length is greater than 1.

**Review by alharris-at — APPROVED** [review]

**Review by AaronZyLee — APPROVED** [review]

**Pull request merged**